[CI] JoinProcessorTests.testJoinWithTargetField fails with NumberFormatException

* Build failure link:  https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.6+multijob-unix-compatibility/os=oraclelinux/65/console
* Error message: 
```
ERROR   0.05s J1 | JoinProcessorTests.testJoinWithTargetField <<< FAILURES!
   > Throwable #1: java.lang.IllegalArgumentException: [󌃋􄿦󔧒⒙𚷲񇒬󅸯򬝙󃪼񾈠] is not an integer, cannot be used as an index as part of path [E.󌃋􄿦󔧒⒙𚷲񇒬󅸯򬝙󃪼񾈠.񛒌󝍌󂆾񟷥񒰒𾲎ࠪ]
   > 	at __randomizedtesting.SeedInfo.seed([AA71D1A0B4BBE8E8:53DA4748DE8256B7]:0)
   > 	at org.elasticsearch.ingest.IngestDocument.setFieldValue(IngestDocument.java:461)
  2> NOTE: leaving temporary files on disk at: /var/lib/jenkins/workspace/elastic+elasticsearch+5.6+multijob-unix-compatibility/os/oraclelinux/modules/ingest-common/build/testrun/test/J1/temp/org.elasticsearch.ingest.common.JoinProcessorTests_AA71D1A0B4BBE8E8-001
   > 	at org.elasticsearch.ingest.IngestDocument.setFieldValue(IngestDocument.java:419)
   > 	at org.elasticsearch.ingest.common.JoinProcessor.execute(JoinProcessor.java:71)
   > 	at org.elasticsearch.ingest.common.JoinProcessorTests.testJoinWithTargetField(JoinProcessorTests.java:129)
   > 	at java.lang.Thread.run(Thread.java:748)
   > Caused by: java.lang.NumberFormatException: For input string: "󌃋􄿦󔧒⒙𚷲񇒬󅸯򬝙󃪼񾈠"
   > 	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
   > 	at java.lang.Integer.parseInt(Integer.java:580)
   > 	at java.lang.Integer.parseInt(Integer.java:615)
   > 	at org.elasticsearch.ingest.IngestDocument.setFieldValue(IngestDocument.java:459)
   > 	... 40 more
```

* Reproduction line: 
  * gradle :modules:ingest-common:test -Dtests.seed=AA71D1A0B4BBE8E8 -Dtests.class=org.elasticsearch.ingest.common.JoinProcessorTests -Dtests.method="testJoinWithTargetField" -Dtests.security.manager=true -Dtests.locale=el-GR -Dtests.timezone=Asia/Ust-Nera (This _does_ reproduce for me on 5.6 branch, a09b0345785eff16d00d96979d15ea8411297712)

--- a/modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/JoinProcessorTests.java
+++ b/modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/JoinProcessorTests.java
@@ -124,7 +124,7 @@ public class JoinProcessorTests extends ESTestCase {
             }
         }
         String fieldName = RandomDocumentPicks.addRandomField(random(), ingestDocument, fieldValue);
-        String targetFieldName = RandomDocumentPicks.randomFieldName(random());
+        String targetFieldName = fieldName + randomAlphaOfLength(5);
         Processor processor = new JoinProcessor(randomAlphaOfLength(10), fieldName, separator, targetFieldName);
         processor.execute(ingestDocument);
         assertThat(ingestDocument.getFieldValue(targetFieldName, String.class), equalTo(expectedResult));

